### PR TITLE
Upgrade gstreamer dependency 2

### PR DESCRIPTION
**Pull request by ceyusa**
Rebase and update of https://github.com/servo/media/pull/393

Just to test CI

### CODE REVIEW

File: backends/gstreamer/lib.rs [hunk 1]
```diff
@@ -103,7 +103,17 @@ impl GStreamerBackend {
             })
             .unwrap();
 
-        thread::spawn(|| glib::MainLoop::new(None, false).run());
+        // from GStreamer 1.19.1 to 1.22.7 glib's MainLoop is required
+        {
+            fn make_version(major: u32, minor: u32, micro: u32) -> u32 {
+                major << 29 | minor << 22 | micro << 12
```
Comment (sdroege) [inline backends/gstreamer/lib.rs:109]: Where do these shifts come from? So odd numbers :D

File: backends/gstreamer/lib.rs [hunk 2]
```diff
@@ -103,7 +103,17 @@ impl GStreamerBackend {
...
+            }
+            let (major, minor, micro, _) = gst::version();
+            let gst_version = make_version(major, minor, micro);
+            if  gst_version >= make_version(1, 19, 1) && gst_version <= make_version(1, 22, 7) {
```
Comment (sdroege) [inline backends/gstreamer/lib.rs:113]: FWIW, you can also just use the `Ord` impl on tuples

File: backends/gstreamer/lib.rs (hunk 1 above)
Comment (ceyusa) [reply backends/gstreamer/lib.rs:109]: from vulkan headers  ;)

File: backends/gstreamer/lib.rs (hunk 2 above)
Comment (ceyusa) [reply backends/gstreamer/lib.rs]: rust is so lovely :)

**Review by rafaelcaricio — APPROVED** [review]

**Comment by ceyusa** [comment]
closing as it's not required anymore...

**Pull request closed without merging**